pkg/autoscaler/async_scaler.go

```diff
@@ -52,7 +52,7 @@ func (s *AsyncScaler) Scale(apiName string, request int32) error {
 		return errors.ErrorUnexpected("unable to find k8s deployment", apiName)
 	}
 
-	current := deployment.Status.Replicas
+	current := *deployment.Spec.Replicas
```

Comment:
although it's probably not going to happen, we should check if the `Spec.Replicas` is nil to avoid panics

Reply:
same applies to the rest of the PR